src/verilog/verilog_preprocessor.cpp

```diff
@@ -591,9 +604,8 @@ void verilog_preprocessort::directive()
 
       if(it==defines.end())
       {
-        error().source_location = source_location;
-        error() << "unknown preprocessor directive \"" << text << "\"" << eom;
-        throw 0;
+        throw verilog_preprocessor_errort()
+          << "unknown preprocessor directive \"" << text << "\"";
```

Comment:
I accept that this satisfies RAII, but I am still wondering whether a constructor that takes a `std::string` wouldn't be the better way forward.

Reply:
It would be a bit weird, e.g., in this case you would have
`throw verilog_preprocessor_errort("unknown preprocessor directive \"")
          << text << "\"";`

Reply:
I would have used + instead of << and have passed the entire string to the constructor. But it’s just a matter of taste, no big deal.

Reply:
Yes, but then you'll want numbers, tokens (which have a << operator, but no +), etc.